tests/torch_datasets/test_process_and_combine.py

```diff
@@ -0,0 +1,143 @@
+import pytest
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from unittest.mock import MagicMock
+
+from ocf_data_sampler.constants import SAT_MEANS, SAT_STDS, NWP_MEANS, NWP_STDS, EPSILON
+from ocf_data_sampler.numpy_batch.satellite import SatelliteBatchKey
+from ocf_data_sampler.numpy_batch.nwp import NWPBatchKey
+from ocf_data_sampler.torch_datasets.process_and_combine import (
+    process_and_combine_datasets,
+    merge_dicts,
+    fill_nans_in_arrays,
+    compute,
+)
+
+
+@pytest.fixture(scope="module")
+def da_sat_like():
+    """ Create dummy satellite DataArray """
+    data = np.random.rand(3, 4, 4).astype(np.float32)
+    coords = {
+        "time_utc": pd.date_range("2023-01-01", periods=3),
+        "x_geostationary": np.arange(4),
+        "y_geostationary": np.arange(4),
+    }
+    return xr.DataArray(
+        data,
+        coords=coords,
+        dims=["time_utc", "y_geostationary", "x_geostationary"],
+    )
+
+
+@pytest.fixture(scope="module")
+def da_nwp_like():
+    """ Create dummy NWP DataArray """
+    data = np.random.rand(3, 3, 4, 4).astype(np.float32)
+    coords = {
+        "init_time_utc": pd.date_range("2023-01-01", periods=3),
+        "step": pd.timedelta_range("0h", periods=3, freq="1h"),
+        "x_osgb": np.arange(4),
+        "y_osgb": np.arange(4),
+    }
+    return xr.DataArray(
+        data,
+        coords=coords,
+        dims=["init_time_utc", "step", "y_osgb", "x_osgb"],
```

Comment:
This is strictly not the data format that `process_and_combine_datasets()` expects. It expects that the NWP data will have been passed through the [`select_time_slice_nwp()` function](https://github.com/openclimatefix/ocf-data-sampler/blob/main/ocf_data_sampler/select/select_time_slice.py#L68) function which changes the dimensions from `["init_time_utc", "step", "channel", "x", "y"]` to `["target_time_utc", "channel", "x", y"]`.

Although admittedly the `process_and_combine_datasets()` doesn't really check this since it is only ever supposed to be used as part of the pipeline where `select_time_slice_nwp()` has been run

Reply:
Updated the test to ensure that NWP data is passed through the select_time_slice_nwp function initially before process and combine is called - would appreciate your feedback on this section if all OK

Reply:
For now I think I'd bypass the `select_time_slice_nwp()` function here and just update this fixture. It means this test is independent of errors in `select_time_slice_nwp()`. 

something like:

```
    data = np.random.rand(3, 3, 4, 4)
    coords = {
        "target_time_utc": pd.date_range("2023-01-01 12:00", freq="1h", periods=3),
        "channel": ["t", "dswrf", "lcc"],
        "x_osgb": np.arange(4),
        "y_osgb": np.arange(4),
    }
    xr.DataArray(
        data,
        coords=coords,
        dims=["target_time_utc", "channel", "y_osgb", "x_osgb"],
    )
```